components/RPC-Connection.jsx

```diff
@@ -117,6 +118,32 @@ async function syncData(network, path, setReturnValue) {
 				chainValue = api.toString();
 				break;
 			case "query":
+				switch (pathParameters[2]) {
```

Comment:
I don't think we should rely on this condition at such a low level in the RPC component.  Every time the component is instantiated elsewhere in the docs and references a `validators` query, this logic will run.  That being said I think all the logic below here is great.  I think moving this to it's own component makes the most sense to keep the RPC component as general as possible.

I made a confusing comment yesterday about adding parameters to the switch statement.  I thought you were trying to use the RPC component to make a query that required you to pass inputs parameters to the function, such as 

```js
api.query.getBalanceFor(SOME_WALLET_ID)
```

Right now we don't support this because there is no way to pass `SOME_WALLET_ID`.  So I was thinking you were trying to extend the RPC component to accept these input parameters, such as 

```js
 <RPC network="polkadot" path="api.query.getBalanceFor" parameters="[SOME_WALLET_ID]"  /> 
```

If you had this functionality you could then use several `<RPC />` components in conjunction to do the calculations below, such as

```js
a = <RPC network="polkadot" path="api.query.session.validators"  /> 
b = <RPC network="polkadot" path="api.query.staking.currentEra"  /> 
c = <RPC network="polkadot" path="api.query.getBalanceFor" parameters="[b, a]"  />
```

This sudo code could be accomplished by creating a new parent component specific to this functionality, that wraps the child components outlined above.  When all the children have valid values you could raise an event that notifies the parent that the display value is ready to be updated.

```js
<MinimumStake>
    <RPC network="polkadot" path="api.query.session.validators"  /> 
    <RPC network="polkadot" path="api.query.staking.currentEra"  /> 
    <RPC network="polkadot" path="api.query.getBalanceFor" parameters="[b, a]"  />
</MinimumStake>
```

Now that I see what you meant, I wouldn't worry about the approach I just described.  It would probably be a pain to synchronize the child components, especially if some have to wait for others to complete before running etc.  I would just create a new component and copy the logic from below into it.

The RPC component is good for retrieving results from a single single endpoint and has the ability to post-process the results using a filter function.  Anything that is more complex probably requires a new component, even if that new component wraps or extends RPC through inheritance.

Sorry for the long-winded reply but I wanted to try and explain my thinking from yesterday and why it probably pushed you in this direction :grin:   Good conversation for our meeting today!

Reply:
Yeah, makes sense. I'll move this to a dedicated component.